qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -151,12 +156,13 @@ def _submit_job(
                 job_name=job_name,
                 job_share_level=job_share_level)
         except ApiError as ex:
-            raise IBMQBackendError('Error submitting job: {}'.format(str(ex)))
+            raise IBMQBackendApiError('Error submitting job: {}'.format(str(ex)))
 
         # Error in the job after submission:
         # Transition to the `ERROR` final state.
         if 'error' in submit_info:
-            raise IBMQBackendError('Error submitting job: {}'.format(str(submit_info['error'])))
+            raise IBMQBackendApiError(
```

Comment:
Probably need to do some testing to see if this path is truly api error or job error (invalid job, no credit, etc).

Reply:
Yeah, I'll look more into it!